sklego/meta/hierarchical_predictor.py

```diff
@@ -378,10 +399,15 @@ def _predict_estimators(self, X, method_name):
 
     def _fit_single_estimator(self, grp_frame):
         """Shortcut to fit an estimator on a single group"""
-        _X = nw.to_native(grp_frame.drop([*self.groups_, self._TARGET_NAME]))
+        _X = nw.to_native(grp_frame.drop([*self.groups_, self._TARGET_NAME, self._SAMPLE_WEIGHT_NAME]))
         _y = nw.to_native(grp_frame[self._TARGET_NAME])
-
-        return clone(self.estimator).fit(_X, _y)
+        
+        args = [_X, _y]
+        if self.estimator_supports_sample_weight and self.has_sw_:
```

Comment:
`and self.has_sw_` can probably be skipped - passing all ones should behave the same as not passing any sample weight (since the estimator supports them)